splunk_connect_for_snmp_poller/manager/hec_sender.py

```diff
@@ -215,31 +215,33 @@ def extract_additional_properties(fields, metric_name, metric_value, server_conf
     for family in oid_families.keys():
         if metric_name.startswith("sc4snmp." + family):
             stripped = metric_name[: metric_name.index("_")]
-
             input_text = metric_name[metric_name.index("_") + 1 :]  # noqa: E203
 
-            entries = oid_families[family][enricher_additional_varbinds]
-            for entry in entries:
-                if "regex" in entry and "names" in entry:
-                    regex = entry["regex"]
-                    names = entry["names"]
-                    names_list = names.split("/")
-
+            entries = multi_key_lookup(
+                oid_families, (family, enricher_additional_varbinds)
+            )
+            if entries:
+                regex_entries = [
+                    entry["regex"] for entry in entries if "regex" in entry
+                ]
+                for regex in regex_entries:
                     result = re.match(regex, input_text)
                     if result:
                         any_regex_matched = True
-                        for index, item in enumerate(names_list):
-                            fields[item] = result.group(index + 1)
+                        for key, value in result.groupdict().items():
+                            fields[key] = value.replace("_", ".")
                         del fields["metric_name:" + metric_name]
                         fields["metric_name:" + stripped] = metric_value
-                        # TODO delete blow debug statement
-                        fields["old_metric_name:" + metric_name] = metric_value
                         continue
+            continue
```

Comment:
Why do we need `continue` here?

Reply:
To end this loop "for family in oid_families.keys():" when we find proper oid_family. There is no need to look check remaining entries. Just an optimisation.